csclient/params/params.go

```diff
@@ -258,6 +262,30 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource holds the metadata about a Resource defined for a charm.
+type Resource struct {
+	Name        string
+	Type        ResourceType
+	Path        string
+	Description string
+	Revision    int
+	Fingerprint string // the hex encoded hash
+	Size        int64
+}
```

Comment:
You're missing Origin.  We have not ruled out supporting more than just OriginStore in the store.  Having the field here would make the information explicit and keep it in line with resource.Resource.  It would also mean we wouldn't have to change the wire format later when Origin is more significant.